sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -165,106 +169,63 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper wit
                 val pushedAggregates = finalTranslatedAggregates.filter(r.pushAggregation)
                 if (pushedAggregates.isEmpty) {
                   aggNode // return original plan node
+                } else if (r.supportCompletePushDown(pushedAggregates.get)) {
+                  // If we can push down the aggregation completely, we need to consider
+                  // pushing down the Limit or Offset operator, so keep the information about
+                  // aggregation push down and push down aggregation later.
+                  sHolder.pushedAggregation = Some(PushedAggregation(
+                    pushedAggregates,
+                    finalResultExpressions,
+                    finalAggregates,
+                    normalizedGroupingExpressions,
+                    aggExprToOutputOrdinal))
+                  sHolder
```

Comment:
The returned scan builder should change its output to `aggNode.output`

Reply:
Got it.